libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/SystemInterceptor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent;
+
+import org.opensearch.javaagent.bootstrap.AgentPolicy;
+
+import java.lang.StackWalker.Option;
+
+import net.bytebuddy.asm.Advice;
+
+/**
+ * {@link System#exit} interceptor
+ */
+public class SystemInterceptor {
```

Comment:
Should we rename this to something more targeted like SystemExitInterceptor or would this be re-usable if we wanted to extend this in the future to getProperty or other System calls?